Frontend/frontend/src/Components/DashboardPayments.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import APIRequest from "../Util/APIRequest";
+import PaymentModule from "./PaymentModule";
+
+
+export default class DashboardPayments extends Component {
+    constructor() {
+        super();
+
+        this.state = {
+            payments: [],
+            contributionPayments: []
+        };
+    }
+
+    async componentDidMount() {
+        // Obtain payments associated with the logged in user.
+        await APIRequest.obtainUserPayments().then(data => {
+            data = data.sort((a, b) => a.startDate.localeCompare(b.startDate)).splice(0, Math.min(4, data.length));
```

Comment:
This doesn't consider if data is empty/undefined, such as a new user not having any payments

Reply:
Actually on second thought, let's merge this in and fix the bug in a new PR